packages/client/src/utils.ts

```diff
@@ -222,7 +263,53 @@ export function dataRowsToCsv(
   const rows = flattenedDataRows.map((flattenedDataRow) =>
     header.map((column) => flattenedDataRow[column])
   );
-  const csvRows = [header, ...rows];
+  const transformedHeader = transformHeader
+    ? header.map(transformHeader)
+    : header;
+  const csvRows = [transformedHeader, ...rows];
   const csvLines = csvRows.map(encodeCsvRow);
   return csvLines.join("\n");
 }
+
+/**
+ * Returns true if an observation date falls between the specified startDate and
+ * endDate. Also returns true if neither startDate nor endDate are specified.
+ *
+ * Dates are specified as ISO-8601 (Examples: "2023", "2023-01-01")
+ *
+ * Truncates observation date length to match the start and end date lengths for
+ * respective comparisons.
+ *
+ * Example:
+ *
+ * observationDate = "2024-05-6"
+ * startDate = "2023"
+ * endDate = "2024-05"
+ *
+ * Truncating the observationDate when comparing to startDate gives us "2024"
+ * Truncating the observation date when comparing to endDate gives us "2024-05"
+ * Result is true because:
+ *   truncatedObservationDate <= endDate (i.e., "2024-05 <= "2024-05)
+ *   AND truncatedObservationDate >= startDate (i.e., "2024" >= "2023")
+ *
+ * @param observationDate Observation date string
+ * @param startDate Start date string
+ * @param endDate End date string
+ * @returns boolean
+ */
+export function isDateInRange(
+  observationDate: string,
+  startDate: string | undefined,
+  endDate: string | undefined
+): boolean {
+  if (!startDate && !endDate) {
```

Comment:
nit: i don't think this check is necessary because both of the following checks checks for the presence of a startDate/endDate

Reply:
Removed